control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -535,6 +587,111 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
+	rootCASecret := manifests.RootCASecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(rootCASecret), rootCASecret); err != nil {
+		return fmt.Errorf("cannot get root CA secret: %w", err)
+	}
+
+	// Etcd client secret
+	clientSecret := manifests.EtcdClientSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get client secret: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, clientSecret, func() error {
+		clientSecret.OwnerReferences = ensureHCPOwnerRef(hcp, clientSecret.OwnerReferences)
+		return ReconcileEtcdClientSecret(clientSecret, rootCASecret)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd client secret: %w", err)
+	}
+
+	// Etcd server secret
+	serverSecret := manifests.EtcdServerSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get server secret: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, serverSecret, func() error {
+		serverSecret.OwnerReferences = ensureHCPOwnerRef(hcp, serverSecret.OwnerReferences)
+		return ReconcileEtcdServerSecret(serverSecret, rootCASecret)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd server secret: %w", err)
+	}
+
+	// Etcd peer secret
+	peerSecret := manifests.EtcdPeerSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get peer secret: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, peerSecret, func() error {
+		peerSecret.OwnerReferences = ensureHCPOwnerRef(hcp, peerSecret.OwnerReferences)
+		return ReconcileEtcdPeerSecret(peerSecret, rootCASecret)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd peer secret: %w", err)
+	}
+
+	// Etcd Operator ServiceAccount
+	operatorServiceAccount := manifests.EtcdOperatorServiceAccount(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorServiceAccount), operatorServiceAccount); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get etcd operator service account: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, operatorServiceAccount, func() error {
+		operatorServiceAccount.OwnerReferences = ensureHCPOwnerRef(hcp, operatorServiceAccount.OwnerReferences)
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd operator service account: %w", err)
+	}
+
+	// Etcd operator role
+	operatorRole := manifests.EtcdOperatorRole(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorRole), operatorRole); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
why do we need this Get request here? what is it giving us that is not already handled by CreateOrUpdate?